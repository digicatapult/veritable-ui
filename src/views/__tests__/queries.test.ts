import { expect } from 'chai'
import { describe, it } from 'mocha'
import QueriesTemplates from '../queries.js'
describe('ConnectionTemplates', () => {
  describe('listPage', () => {
    it('should render with no connections', async () => {
      const templates = new QueriesTemplates()
      const rendered = await templates.chooseQueryPage()
      expect(rendered).to.matchSnapshot()
    })
  })
})
