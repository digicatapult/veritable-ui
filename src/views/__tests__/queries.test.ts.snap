// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTemplates listPage should escape html in name 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main connections\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"connections header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"connections list\\"><div class=\\"connections-list-nav\\"><span>Query Management</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"connections list\\"><thead><th><span>Company Name</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"connections-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>&lt;div>I own you&lt;/div></td><td>Type A</td><td><p>Received</p></td><td>Tue Jul 02 2024 15:38:10 GMT+0100 (British Summer Time)</td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr></tbody></table></div></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with no connections 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon active\\" style=\\"--background-image: url(&#34;/public/images/category.svg&#34;)\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon \\" style=\\"--background-image: url(&#34;/public/images/connection.svg&#34;)\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/folder.svg&#34;)\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/notification.svg&#34;)\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/setting.svg&#34;)\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Queries</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><span class=\\"url-separator\\">/</span><a title=\\"Queries\\" href=\\"/queries/new\\">Queries</a></div></div><div class=\\"content main\\"><div class=\\"main-list-page\\" hx-get=\\"/queries/new\\"></div><div><h1 class=\\"query-page-header\\">Queries</h1><hr class=\\"divider\\"/></div><div class=\\"query-container\\"><a class=\\"query-item\\" onclick=\\"location.href='/queries/new/scope-3-carbon-consumption';\\"><h1 class=\\"query-header\\">Scope 3 Carbon Consumption</h1><p class=\\"query-text\\">Creates a query for calculating the total scope 3 carbon consumption for a given product or component.</p></a><a class=\\"query-item\\" onclick=\\"location.href='#';\\"><h1 class=\\"query-header\\">Product Provenance</h1><p class=\\"query-text\\">Creates a query to assert that a product and its components are not coming from blacklisted areas.</p></a><a class=\\"query-item\\" onclick=\\"location.href='#';\\"><h1 class=\\"query-header\\">Aerospace Grade</h1><p class=\\"query-text\\">Creates a query to certify all components of a product are manufactured according to aerospace grade standards such as AS9100.</p></a><a class=\\"query-item\\" onclick=\\"location.href='#';\\"><h1 class=\\"query-header\\">ISO 9001 Compliance</h1><p class=\\"query-text\\">Creates a query to verify that all the companies involved in manufacturing a given product are ISO9001 certified.</p></a></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with single query 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main connections\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"connections header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"connections list\\"><div class=\\"connections-list-nav\\"><span>Query Management</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"connections list\\"><thead><th><span>Company Name</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"connections-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>Company A</td><td>Type A</td><td><p>Received</p></td><td>Tue Jul 02 2024 15:38:10 GMT+0100 (British Summer Time)</td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr></tbody></table></div></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with single qury 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main connections\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"connections header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"connections list\\"><div class=\\"connections-list-nav\\"><span>Query Management</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"connections list\\"><thead><th><span>Company Name</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"connections-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"connections-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>Company A</td><td>Type A</td><td><p>Received</p></td><td>Tue Jul 02 2024 15:37:09 GMT+0100 (British Summer Time)</td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr></tbody></table></div></div></div></main></body></html>"`;
